.github/workflows/build_test_release.yml

```diff
@@ -0,0 +1,101 @@
+name: Build, Test, and Release workflows
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    tags: [ 'v*.*.*' ]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - name: Install build dependencies
+        run: |
+          python -m pip install build --user
+      - name: Build package
+        run: |
+          python -m build
+      - name: Upload package
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+          retention-days: 1
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install package dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ".[test]"
```

Comment:
For this "release" workflow, we might want to explicitly install one of the built packages (in `/dist`)

Reply:
I updated the new `test.yml` to install from `dist/`. It runs on push as well, but its not gating. Probably good to rely on tests to pass before running the release workflow?